src/gsy_e/models/strategy/market_agents/two_sided_engine.py

```diff
@@ -50,8 +50,6 @@ def __repr__(self):
     def _forward_bid(self, bid):
         if bid.buyer == self.markets.target.name:
             return None
-        if self.owner.name == self.markets.target.name:
-            return None
```

Comment:
These lines were removed as a fix for the market agents' integration tests, I couldn't see the rational behind their existence in the first place.
FYI @spyrostz @hannesdiedrich

Reply:
Are you certain that the integration tests pass with this change, and that the same bid does not get recreated to the same area? I need to run this part of the code to tell you with certainty, but I vaguely recall that this condition ensured that the bid will not be recreated on the same market. For instance, if this agent/engine is controlled by market House 1 and the forward_bid is called from the engine higher_market -> House 1 with a bid that has already been created in House 1, then the same bid would be recreated on the House 1 market. Again, this is from my memory, so take this with a grain of salt. 
Could be that more code has been added meanwhile that makes this condition useless, however I think that it is good to have a second layer of protection here. Why did this condition need to be removed in the first place in order for integration tests to pass ?

Reply:
@spyrostz AFAIK the source and target markets are actual markets, not agents; therefore their names shouldn't have MA... prefixes before their names.
Looking again at the lines I removed:
```python
if self.owner.name == self.markets.target.name:
```
the owner name used to have a prefix (MA House 1 for instance) whereas the `self.markets.target.name` shouldn't have such prefix (even if the target market is House 1, this condition will still be True). 
This is why I believe that this condition wasn't actually needed